### ISSUE TITLE
secret not set when running job on PR

### ISSUE THREAD

**Issue by cjw296** [travis-ci/travis-ci#10117]
I have a secret that provides a coveralls repo token, as can be seen here:

https://github.com/cjw296/testfixtures/blob/6e8807543b804946aba58e2c9e92f5bdc3656a57/.travis.yml#L13

However, on this PR, it does not appear to have been set in the final "Coverge" step:
https://github.com/Simplistix/testfixtures/pull/88/files#diff-354f30a63fb0907d4ad57269548329e3

Here's the failing job:
https://travis-ci.org/Simplistix/testfixtures/jobs/428481452#L445

Contrast that with this passing on master:
https://travis-ci.org/Simplistix/testfixtures/jobs/428484004

Is this a bug or am I doing something wrong?

**Comment by BanzaiMan** [comment]
This is expected behavior. PRs coming from forks do not have access to secrets. For more information, lease consult https://docs.travis-ci.com/user/pull-requests#pull-requests-and-security-restrictions

> Travis CI makes encrypted variables and data available only to pull requests coming from the same repository. These are considered trustworthy, as only members with write access to the repository can send them.

Thanks.

**Comment by cjw296** [comment]
@BanzaiMan - hmm, that's frustrating. Please see https://github.com/lemurheavy/coveralls-public/issues/1139 for why I (and others) are looking to do this. Is there any way to "whitelist" some encrypted variables to be able to use them in jobs running against PRs from other repos?

I'm open to suggestions here too...

**Comment by cjw296** [comment]
@BanzaiMan - any follow up on this? Still looking for a solution to this...

**Comment by BanzaiMan** [comment]
Sorry, but there is no update. If you allow access to your secrets during the PR builds, it is as good as being public. Your nefarious PR can access it, and send it somewhere you don't know about without your knowledge.

**Comment by cjw296** [comment]
@BanzaiMan - Okay, so what now? I tried to send this to your new community thing, but it just vanished and I got a "Account temporarily on hold" so trying again here:

How can I share coverage information between build stages in PRs from forks?

I have a lot of projects that use matrices and build stages to build across different python versions and a stage that checks whether coverage remains at 100%. I need this to work on branches, such as master, and on PRs, including those from forks.

I had this almost working using coveralls.io, but because encrypted environment variables do not work with PRs from forks, I can't tell the coveralls all the parallel builds are done.

The advice seems to be to use S3, but this looks like this has the same problem in that encrypted environment variables do not work with PRs from forks.

I've tried to (ab)use build caches to get this information across, but this build stage that checks coverage only sees cache files for the python version used to run it.

I can't see any way through this, what should I do?
